app/src/main/java/com/example/weather/screen/map/MapFragment.kt

```diff
@@ -41,26 +80,84 @@ class MapFragment private constructor() : BaseFragment<FragmentMapBinding>(), On
         mapFragment.getMapAsync(this)
 
         viewBinding.currentLocationButton.setOnClickListener {
-            mGoogleMap?.clear()
-            zoomOnMap(LatLng(mCurrentLatitude, mCurrentLongitude))
-            addMarker(LatLng(mCurrentLatitude, mCurrentLongitude))
+            mPresenter?.getWeather(mCurrentLatitude, mCurrentLongitude)
+            moveToLocation(mCurrentLatitude, mCurrentLongitude)
         }
+
+        viewBinding.mapSearch.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String?): Boolean {
+                val location = viewBinding.mapSearch.query.toString()
+                var addressList: MutableList<Address>? = null
+                val geocoder = context?.let { Geocoder(it) }
+                try {
+                    addressList = geocoder?.getFromLocationName(location, 1)
+                } catch (e: IOException) {
+                    println(e)
+                }
+                if (addressList != null) {
+                    if (addressList.isNotEmpty()) {
+                        mSelectedLatitude = addressList[0].latitude
+                        mSelectedLongitude = addressList[0].longitude
+                        mPresenter?.getWeather(mSelectedLatitude, mSelectedLongitude)
+                        moveToLocation(mSelectedLatitude, mSelectedLongitude)
+                    } else {
+                        Toast.makeText(
+                            context,
+                            getString(R.string.location_not_exist),
+                            Toast.LENGTH_SHORT
+                        ).show()
+                    }
+                }
+                return false
+            }
+
+            override fun onQueryTextChange(newText: String?): Boolean {
+                return false
+            }
+        })
+
         viewBinding.favoriteLocationButton.setOnClickListener {
             // TODO implement later
         }
         viewBinding.icBack.setOnClickListener {
             goBackFragment()
         }
+
         viewBinding.layoutWeatherMap.cardViewMap.setOnClickListener {
-            // TODO implement later
+            activity?.let {
+                mWeather?.let { weather -> DetailFragment.newInstance(weather) }?.let { detailFragment ->
+                    (it as AppCompatActivity).replaceFragmentToActivity(
+                        it.supportFragmentManager,
+                        detailFragment,
+                        R.id.container
+                    )
+                }
+            }
         }
+
         viewBinding.layoutWeatherMap.icStar.setOnClickListener {
-            // TODO implement later
+            if (viewBinding.layoutWeatherMap.icStar.drawable.constantState?.hashCode() == ContextCompat.getDrawable(
```

Comment:
```suggestion
            if (viewBinding.layoutWeatherMap.icStar.drawable.constantState?.hashCode() == ContextCompat.getDrawable(
```
Ca này mình lấy biến ``mWeather`` mà check chứ em không dựa vào UI như này 😮‍💨 

Tìm cách chặn user click liên tục nữa em nhé!